source/server/admin/admin.cc

```diff
@@ -221,7 +221,9 @@ AdminImpl::AdminImpl(const std::string& profile_path, Server::Instance& server,
                         "Render text_readouts as new gaugues with value 0 (increases Prometheus "
                         "data size)"},
                        {ParamDescriptor::Type::String, "filter",
-                        "Regular expression (Google re2) for filtering stats"}}),
+                        "Regular expression (Google re2) for filtering stats"},
+                       {ParamDescriptor::Type::String, "histogram_mode",
+                        "Format to report histograms as; One of (histogram, summary, none)"}}),
```

Comment:
consider also 'detail' which is a new mode available in the admin UI and the histogram API, showing the full detail of what the underlying histogram detail contains. The disadvantage of that relative to the "cumulative" or "disjoint" is that multiple Envoys may choose different buckets based on the data they see, so Prom might not be able to aggregate them.

Moreover, it's a little confusing that you are defining a new histogram enum that's a lot like the one `/stats` already has: 
```
  Admin::ParamDescriptor histogram_buckets{Admin::ParamDescriptor::Type::Enum,
                                           "histogram_buckets",
                                           "Histogram bucket display mode",
                                           {"cumulative", "disjoint", "detailed", "none"}};
```
see source/server/admin/stats_handler.cc:180 which is similar to this but different. And what's super confusing is that /stats?format=prometheus basically does an internal redirect to /stats/prometheus. See source/server/admin/stats_handler.cc:82. Should we just re-use that param-descriptor so this is all compatible across formats?

In that case we'd just use `histogram_buckets_mode` for consistency with what's in the product now.

Reply:
I want to point out also that `"none"` above really means `"summary"` and is the default setting for that field.

It occurs to me know that the name `"none"` is misleading (this was my fault for not thinking that thorugh) and we should really rename that `"summary"` and let `"none"` be a permanently deprecated synonym.

Reply:
and one thing that's a little annoying, when merging these overlapping params, is that the default for histogram _buckets_mode" for non-prom stats is `"none"` (aka 'summary') and the default I guess for Prom is effectively `cumulative` I think.

Reply:
I thought about this a little more and I don't see an issue. If unspecified we can give prom a different behavior than non-prom (text,html,json) but they should still be the same set of options.

Reply:
Had gone through a similar thought process initially around whether to just use `none`, but decided to err on the side of not changing existing defaults/behavior 😅 I've updated the PR to reflect what I believe is the desired behavior; only emit summary metrics on an explicitly set `histogram_buckets=none` query param.

It feels a little off to be checking things about `params.query_` in the prometheus handler, but that seemed to be the least invasive way to get the desired behavior. An alternative would be to add an explicit `Utility::HistogramBucketsMode::Unset` variant, and then different handlers can have different default behaviors, although I'm not sure if that would be even more confusing to users.

Reply:
agreed -- let's centralize the handling of the inconsistent defaults in stats_params.cc instead.